cargo-workspaces/src/utils/version.rs

```diff
@@ -151,19 +151,25 @@ impl VersionOpt {
             )?;
         }
 
-        let workspace_root = metadata.workspace_root.join("Cargo.toml");
-        fs::write(
-            &workspace_root,
-            format!(
-                "{}\n",
-                change_versions(
-                    fs::read_to_string(&workspace_root)?,
-                    "",
-                    &new_versions,
-                    self.exact
-                )?
-            ),
-        )?;
+        if let Some(new_version) = &new_version {
+            let workspace_root = metadata.workspace_root.join("Cargo.toml");
+            let mut workspace_version_map = Map::new();
+
+            workspace_version_map.insert("".to_string(), new_version.clone());
+
+            fs::write(
+                &workspace_root,
+                format!(
+                    "{}\n",
+                    change_versions(
+                        fs::read_to_string(&workspace_root)?,
+                        "",
+                        &workspace_version_map,
```

Comment:
It looks like this patch breaks version updates of inherited dependencies from the workspace.

```toml
# Cargo.toml (root)
[workspace.package]
version = "0.1.0" # <-- this will get updated

[workspace.dependencies]
foo = { path = "./crates/foo", version = "0.1.0" } # <─┐
                                                   #   ├─> these wont
bar = { path = "./crates/bar", version = "0.1.0" } # <─┘
```

```toml
# crates/foobar/Cargo.toml
[package]
version.workspace = true

[dependencies]
foo.workspace = true
bar.workspace = true
```

Reply:
Weird, I was just testing that. Let me check again, some bug might have gotten in during the final edits.

Reply:
The issue is the patch discards the update information from `new_versions` and instead uses a special map of one entry to aid updating the workspace version. The correct approach would've been to append the new entry to the already existing map.

Reply:
Yup, that was the final edit I made, I fixed it now and published 0.2.39